presto-hive/src/main/java/io/prestosql/plugin/hive/HivePartitionManager.java

```diff
@@ -223,7 +223,8 @@ public HiveTableHandle applyPartitionResult(HiveTableHandle handle, HivePartitio
                 partitions.getBucketHandle(),
                 partitions.getBucketFilter(),
                 handle.getAnalyzePartitionValues(),
-                handle.getAnalyzeColumnNames());
+                handle.getAnalyzeColumnNames(),
+                partitionConstraint);
```

Comment:
The Constraint object is only meant to be used by the immediate callees and not intended to be stored or used once the call is complete (it may hold on to unspecified resources under the covers -- e.g., compiled code, etc., or depend on state that is not available once the call completes). Therefore, it can't be stored in the handle for use later on.